### ISSUE TITLE
KVO: crash when removing observer for nested key paths

### ISSUE THREAD

**Issue by triplef** [Microsoft/WinObjC#225]
Assuming a class `Container` defined as follows:

```obj-c
@interface Container : NSObject
@property (nonatomic, retain) Container *child;
@end
```

The last line in the following code will cause a crash in `Foundation.dll!KVONotifier::_removeSubscriber`:

```obj-c
Container *parent = [[Container alloc] init];
[parent addObserver:self forKeyPath:@"child.child" options:0 context:nil];
[parent removeObserver:self forKeyPath:@"child.child"]; // crash
```

**Comment by DHowett-MSFT** [comment]
We've got some changes on the way that resolve this and a couple other KVO issues, including #224 and #226.

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)

**Comment by Ramu-msft** [comment]
Fix is available in 0.1 Preview (November 12, 2015)
